alert-database/src/test/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessorTest.java

```diff
@@ -0,0 +1,251 @@
+
+        Mockito.when(configurationAccessor.getProviderConfigurationByName(Mockito.any())).thenReturn(Optional.of(configurationModel));
+        Mockito.when(providerProjectRepository.findByProviderConfigId(Mockito.any())).thenReturn(List.of(providerProjectEntity));
+
+        DefaultProviderDataAccessor providerDataAccessor = new DefaultProviderDataAccessor(providerProjectRepository, null, null, configurationAccessor);
+        List<ProviderProject> providerProjectList = providerDataAccessor.getProjectsByProviderConfigName("test-providerConfigName");
+
+        assertEquals(1, providerProjectList.size());
+        ProviderProject providerProject = providerProjectList.get(0);
+        assertEquals(name, providerProject.getName());
+        assertEquals(description, providerProject.getDescription());
+        assertEquals(href, providerProject.getHref());
+        assertEquals(projectOwnerEmail, providerProject.getProjectOwnerEmail());
+    }
+
+    @Test
+    public void getProjectsByProviderConfigNameEmptyTest() throws Exception {
+        ConfigurationAccessor configurationAccessor = Mockito.mock(ConfigurationAccessor.class);
+        ProviderProjectRepository providerProjectRepository = Mockito.mock(ProviderProjectRepository.class);
+
+        Mockito.when(configurationAccessor.getProviderConfigurationByName(Mockito.any())).thenReturn(Optional.empty());
+
+        DefaultProviderDataAccessor providerDataAccessor = new DefaultProviderDataAccessor(providerProjectRepository, null, null, configurationAccessor);
+        List<ProviderProject> providerProjectList = providerDataAccessor.getProjectsByProviderConfigName("test-providerConfigName");
+
+        assertEquals(0, providerProjectList.size());
+    }
+
+    @Test
+    public void getProjectsByProviderConfigIdTest() {
+        ProviderProjectEntity providerProjectEntity = new ProviderProjectEntity(name, description, href, projectOwnerEmail, 1L);
+
+        ConfigurationAccessor configurationAccessor = Mockito.mock(ConfigurationAccessor.class);
+        ProviderProjectRepository providerProjectRepository = Mockito.mock(ProviderProjectRepository.class);
+
+        Mockito.when(providerProjectRepository.findByProviderConfigId(Mockito.any())).thenReturn(List.of(providerProjectEntity));
+
+        DefaultProviderDataAccessor providerDataAccessor = new DefaultProviderDataAccessor(providerProjectRepository, null, null, configurationAccessor);
+        List<ProviderProject> providerProjectList = providerDataAccessor.getProjectsByProviderConfigId(1L);
+
+        assertEquals(1, providerProjectList.size());
+        ProviderProject providerProject = providerProjectList.get(0);
+        assertEquals(name, providerProject.getName());
+        assertEquals(description, providerProject.getDescription());
+        assertEquals(href, providerProject.getHref());
+        assertEquals(projectOwnerEmail, providerProject.getProjectOwnerEmail());
```

Comment:
I've seen this a couple times. You could abstract it into a private method like you did in the other test.